scrapy/extensions/feedexport.py

```diff
@@ -32,8 +32,9 @@
 class IFeedStorage(Interface):
     """Interface that all Feed Storages must implement"""
 
-    def __init__(uri):
-        """Initialize the storage with the parameters given in the URI"""
+    def __init__(uri, feed):
```

Comment:
What do you think about calling it feed_options?

* for feed argument I'd expect some instance of Feed, whatever is it
* feed_settings is not god, because this dict is not a Settings instance